blockly-core/core/ui/contract_editor/contract_editor.js

```diff
@@ -152,37 +168,146 @@ Blockly.ContractEditor.prototype.create_ = function() {
+          newY += blockSplitMargin;
+
+          callText.style.top = newY + 'px';
+          callText.style.left = EXAMPLE_BLOCK_MARGIN_LEFT + 'px';
+
+          resultText.style.top = newY + 'px';
+          resultText.style.left = verticalMidlineOffset + EXAMPLE_BLOCK_MARGIN_LEFT + 'px';
+
+          newY += callText.offsetHeight;
+
+          newY += blockSplitMargin;
+
+          topHorizontalLine.setAttribute('transform', 'translate(' + 0 + ',' + newY + ')');
+          topHorizontalLine.setAttribute('width', this.getFullWidth());
+
+          var i = 0;
+          for (; i < this.exampleBlocks.length; i++) {
+            var block = this.exampleBlocks[i];
+            var canReuse = this.exampleViews_.length > i;
+            if (!canReuse) {
+              var newExampleView = new Blockly.ExampleView(this.exampleAreaDiv, examplesTableGroup, function (block) {
+                this.exampleViews_.forEach(function (exampleView) {
+                  exampleView.resetExample_();
+                });
+
+                // TODO(bjordan): Reset main workspace runner esp. if visualizing?
+
+                this.contractSectionView_.setCollapsed(true);
+                return this.testHandler_(block);
+              }.bind(this), function () {
+                this.testResetHandler_();
+              }.bind(this));
+              this.exampleViews_.push(newExampleView);
+            }
+            newY = this.exampleViews_[i].placeExampleAndGetNewY(block, newY, maxWidth,
+              EXAMPLE_BLOCK_MARGIN_LEFT, EXAMPLE_BLOCK_MARGIN_BELOW, this.getFullWidth(), verticalMidlineOffset);
+          }
+        }
+
+        for (var j = this.exampleBlocks.length; j < this.exampleViews_.length; j++) {
+          this.exampleViews_[j].dispose();
+        }
+        this.exampleViews_.length = this.exampleBlocks.length;
+
+        verticalMidline.style.display = exampleSectionVisible ? 'block' : 'none';
+        topHorizontalLine.style.display = exampleSectionVisible ? 'block' : 'none';
+        callText.style.display = exampleSectionVisible ? 'block' : 'none';
+        resultText.style.display = exampleSectionVisible ? 'block' : 'none';
 
-        newY = this.addExampleButton.renderAt(EXAMPLE_BLOCK_MARGIN_LEFT, newY);
+        verticalMidline.setAttribute('height', newY - verticalMidlineY);
+
+        newY += blockSplitMargin;
+
+        this.addExampleButton.style.top = newY + 'px';
+        this.addExampleButton.style.left = EXAMPLE_BLOCK_MARGIN_LEFT + 'px';
+        newY += this.addExampleButton.offsetHeight;
         newY += EXAMPLE_BLOCK_SECTION_MAGIN_BELOW;
+
+        this.exampleAreaDiv.style.height = (newY - currentY) + 'px';
+
         return newY;
       }, this),
       highlightBox: sharedHighlightBox,
       onCollapseCallback: goog.bind(function (isNowCollapsed) {
+        this.exampleAreaDiv.style.display = isNowCollapsed ? 'none' : 'block';
+        examplesTableGroup.style.display = isNowCollapsed ? 'none' : 'block';
```

Comment:
Blockly.ContractEditor.prototype.create_ has gotten to be pretty long. I wonder if we can/should split some things out.